src/btree/bt_ovfl.c

```diff
@@ -109,51 +121,100 @@ __ovfl_cache_col_visible(
  *	row-store: check for a globally visible update.
  */
 static bool
-__ovfl_cache_row_visible(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip)
+__ovfl_cache_row_visible(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 {
-	WT_UPDATE *upd;
-
 	/* Check to see if there's a globally visible update. */
-	for (upd = WT_ROW_UPDATE(page, rip); upd != NULL; upd = upd->next)
-		if (__wt_txn_upd_visible_all(session, upd))
+	for (; upd != NULL; upd = upd->next)
+		 if ((upd->type == WT_UPDATE_STANDARD ||
+		     upd->type == WT_UPDATE_DELETED) &&
```

Comment:
We have to ignore reserved records here because they can't be returned to the application, and so don't mean we won't have to cache the overflow value. I should have called it out in the commit, it's a bug fix -- we'll get into trouble if there's a reserve update here and we don't cache an overflow value because of it.


I've pushed a macro in 7fbb00af9 so adding more update states won't be fragile.